utils/python/CIME/case.py

```diff
@@ -24,13 +24,19 @@
 class Case(object):
 
     def __init__(self, case_root=os.getcwd()):
+        
+        # Init first, if no valid case_root expect fails and tears down object, __del__ expects self._env_files_that_need_rewrite
+        self._env_files_that_need_rewrite = set()
+        
         expect(os.path.isdir(case_root),
```

Comment:
This will be removed in an upcoming commit - when we bring in create_newcase.   If __del__ is called before the case_root is created it should not call flush()